### DISCUSSION TITLE
Disallow extensions to show "What's new" panels

### DISCUSSION THREAD

**Discussion by ratermir** [eclipse-theia/theia#9182] [Ideas]
Some extensions show "What's new in ..." panels every time, when IDE starts. In case of Theia it is every time when Theia reconnects to the server, which may occure quite frequently on non-perfect line. There are situations when I left computer for a certain time and after return I have tens of "What's new" panels opened, which is really annoying 
I know better solution is to contact extension devlopers to stop showing "What's new" completely or disable this globally in the config for the extension or so, but unfortunalely it dows not work (developers don't response, no time etc ...)

So my idea is whether is it possible to disable opening every new panels with title beginning with "What's new in". Usually I don't need to see it at all - I thing idea of these panels is nonsense anyway. 
I know it is a workaround only, but it could work.

Another solution could be to distinguish new IDE browser window from "only reconnection" of existing window. But I don't know whether is it possible and how it would affect extensions

**Comment by vince-fugnitto** [comment]
@ratermir thank you for the discussion, do you have an example plugin that has this behavior, I'd like to take a look at their sources a bit and see what types of API they may use (ex: `webview`). In theory it should be possible to customize the default behavior for your own custom Theia-based application to not have such views, or even control them through a preference (so some of your users can turn the feature on/off).

**Reply by ratermir** [reply]
Just now I use two such plugins: Project manager - https://github.com/alefragnani/vscode-project-manager and the other is OpenApi editor - https://open-vsx.org/extension/42crunch/vscode-openapi.

But there are probably more such extensions since I found more similar questions in various discussions.

**Reply by vince-fugnitto** [reply]
@ratermir both extensions use two different ways to create the '_what's new_' webviews. You might be able to customize the default behavior in your own application to not display such views by extending the framework's default behavior.

**Reply by ratermir** [reply]
that's great, thank you for this info. But I don't know how to achieve it. I have no experiences with Theia development and very small experiences with NodeJS/Typescript/javascript. Can you point me how to do it ?

**Reply by vince-fugnitto** [reply]
> that's great, thank you for this info. But I don't know how to achieve it. I have no experiences with Theia development and very small experiences with NodeJS/Typescript/javascript. Can you point me how to do it ?

Unfortunately I do not have the time to look into the specifics of what is required in order to get it to work, but I can try to point you to some documentation:

- [creating custom applications](https://theia-ide.org/docs/composing_applications)
- [creating custom `@theia` extensions](https://theia-ide.org/docs/authoring_extensions/)

You'll likely need to verify how such webviews are created and think of a strategy on how to discard such views from opening.
I don't believe it is an easy feature to necessarily implement, especially since you mentioned not having much experience yet.

**Reply by ratermir** [reply]
Thank you for these tips. Really it is not necessary to implement - everything works without it. It's for convenience only; closing of tens of unněěded panels is anoying. But it could be a good starting point for me to begin with Theia development :)
